kuzzleio/sdk-go
# Wrap Listeners for android

## What does this PR do ?

Wrap NotificationListener, EventListener and SubscribeListener for android.
First make each pure virtual methods abstract for Java..
Then wrap JSON type when needed

### How should this be manually tested?

Check if it builds:
```sh
docker run --rm -it -v "$(pwd)":/go/src/github.com/kuzzleio/sdk-go android-x86 /custom_build.sh
```
You can then test it in an android project (for x86 for now).
Example of a subscribe with NotificationListener:

```java
Kuzzle k = new Kuzzle("kuzzlehost");
k.getRealtime().subscribe("index", "collection", "{}", new NotificationListener() {
    @Override
    public void onMessage(NotificationResult notificationResult) {
        try {
            Log.e("e", notificationResult.getResult().getContent().toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
});
```
